.github/workflows/push.yaml

```diff
@@ -0,0 +1,93 @@
+#
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-gradle
+
+name: Publish Polaris Server
+
+on:
+  push:
+    branches: [ "GH-635-add-build-workflow" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v4
+      # Get the current date in YYYY-mm-dd format to be used as the image tag
+      - name: Get current date
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+
+      # Configure Gradle for optimal use in GiHub Actions, including caching of downloaded dependencies.
+      # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@16bf8bc8fe830fa669c3c9f914d3eb147c629707 # v4
+        with:
+          # The setup-gradle action fails, if the wrapper is not using the right version or is not present.
+          # Our `gradlew` validates the integrity of the `gradle-wrapper.jar`, so it's safe to disable this.
+          validate-wrappers: false
+
+      - name: Build with Gradle Wrapper
+        run: ./gradlew build
+        env:
+          ORG_GRADLE_PROJECT_mavenUser: ${{ secrets.JFROG_USER }}
+          ORG_GRADLE_PROJECT_mavenPassword: ${{ secrets.JFROG_TOKEN }}
+          ORG_GRADLE_PROJECT_snapshotsRepoURL: ${{ vars.MAVEN_SNAPSHOT_URL }}
+          ORG_GRADLE_PROJECT_releasesRepoURL: ${{ vars.MAVEN_RELEASE_URL }}
+      # The following step is needed to enable multi-platform Docker builds  
+      - uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker-container
+      - name: Docker Login
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ vars.CONTAINER_REGISTRY }}
+          username: ${{ secrets.JFROG_USER }}
+          password: ${{ secrets.JFROG_TOKEN }}
+      - name: Set Docker Metadata
+        uses: docker/metadata-action@v5
+        id: meta
+        with:
+          images: ${{ vars.CONTAINER_REGISTRY }}/${{ github.repository }}
+          tags: |
+            type=sha
+            type=raw,value=${{ steps.date.outputs.date }}
+      - name: Docker Build
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          platforms: linux/amd64,linux/arm64
+          tags: ${{ steps.meta.outputs.tags }}
+          cache-from: type=registry,ref=ghcr.io/${{ github.repository }}/buildcache
```

Comment:
I don't think that you need either caching entry for Polaris.  Caching is a new feature for data-processing because that repo builds so many things.

Reply:
Agreed. Mao setup the caching, so I don't know exactly how it works. It was added because there were 5 image builds that used a common intermediate image in JFrog. Adding the cache reduced the number of pulls from JFrog, which reduced our bill. Here's the [documentation](https://docs.docker.com/build/ci/github-actions/cache/#registry-cache).